WooCommerce/Classes/ViewRelated/Orders/Order Filters/FilterOrderListViewModel.swift

```diff
@@ -87,25 +97,35 @@ final class FilterOrderListViewModel: FilterListViewModel {
         dateRangeFilterViewModel = OrderListFilter.dateRange.createViewModel(filters: filters, allowedStatuses: allowedStatuses)
         productFilterViewModel = OrderListFilter.product(siteID: siteID).createViewModel(filters: filters, allowedStatuses: allowedStatuses)
         customerFilterViewModel = OrderListFilter.customer(siteID: siteID).createViewModel(filters: filters, allowedStatuses: allowedStatuses)
+        salesChannelFilterViewModel = OrderListFilter.salesChannel.createViewModel(filters: filters, allowedStatuses: allowedStatuses)
 
         self.siteID = siteID
         self.stores = stores
         self.analytics = analytics
 
         shouldShowHistory = featureFlagService.isFeatureFlagEnabled(.filterHistoryOnOrderAndProductLists)
-        filterTypeViewModels = [orderStatusFilterViewModel, dateRangeFilterViewModel, customerFilterViewModel, productFilterViewModel]
+        filterTypeViewModels = [orderStatusFilterViewModel,
+                                dateRangeFilterViewModel,
+                                customerFilterViewModel,
+                                productFilterViewModel]
+
+        if featureFlagService.isFeatureFlagEnabled(.pointOfSaleOrdersi2) {
+            filterTypeViewModels.append(salesChannelFilterViewModel)
+        }
```

Comment:
If this is the only reason the `filterTypeViewModels` needs to be a var, it would be better to make a local var, and then set the property with it when you're done constructing it.